src/prescriptions/prescription-tab-lists.component.tsx

```diff
@@ -30,6 +31,15 @@ const PrescriptionTabLists: React.FC = () => {
     },
   ];
 
+  // debounce: delay the search term update so that a search isn't triggered on every single keystroke
+  useEffect(() => {
+    const debounceFn = setTimeout(() => {
+      setSearchTerm(searchTermUserInput);
+    }, 500);
```

Comment:
We make sure we wait 500ms before updating the search term prop so that we don't trigger a new search on *every* keystroke